lib/sales_analyst.rb

```diff
@@ -80,11 +86,129 @@ def standard_deviation(values, mean)
     std_dev.round(2)
   end
 
+  def average_invoices_per_merchant
+    average = all_invoices.length.to_f / all_merchants.length
+    average.round(2)
+  end
+
+  def num_of_invoices_per_merchant
+    all_merchants.each_with_object({}) do |merchant, total_per_merchant|
+      total_invoices = all_invoices.count do |invoice|
+        invoice.merchant_id == merchant.id
+      end
+      total_per_merchant[merchant] = total_invoices
+    end
+  end
+
+  def average_invoices_per_merchant_standard_deviation
+    mean = average_invoices_per_merchant
+    invoice_counts = num_of_invoices_per_merchant.values
+    std_dev = standard_deviation(invoice_counts, mean)
+  end
+
+  def top_merchants_by_invoice_count
+    mean = average_invoices_per_merchant
+    std_dev = average_invoices_per_merchant_standard_deviation
+
+    z = standard_deviations_of_mean(mean, std_dev, 2)
+
+    merchants = []
+    num_of_invoices_per_merchant.each_pair do |merchant, invoice_count|
+      merchants << merchant if invoice_count >= z
+    end
+    merchants
```

Comment:
⛏ Could you refactor this method like #golden_items to make it more concise?

Reply:
I'll dig into that